File: qkfcl9987/level2/큰 수 만들기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string number, int k) {
    string answer = "";
    vector<int> temp;
    int digits = number.size() - k;

    for (int i = digits; 0 < i; i--)
    {
        int dump = 0;
        int index = 0;
        //높은 숫자 뽑기
        for (int j = 0; j < number.size() - i + 1; j++)
        {
            if (dump < number[j])
            {
                dump = number[j];
                index = j;
            }
        }
        //높은 숫자까지 전부 공백
        number.replace(0, index + 1, " ");
        temp.push_back(dump);
    }
    for (int i = 0; i < temp.size(); i++)
        answer += temp[i];
    return answer;
}

File: sejin/highScoreKit/큰수 만들기.py
def solution(number, k):
    number = list(number)
    cnt = 0
    while cnt< k:
        for i in range(0, len(number)-1):
            if int(number[i]) < int(number[i+1]):
                print(number[i])
                del number[i]
                cnt += 1
                break
    answer = "".join(number)
    return answer